R/make_ReRingTable.R

```diff
@@ -0,0 +1,79 @@
+#Use N to create new colnames (e.g. RING3, RING4)
+
+#' Make a summary table containing all ring numbers of re-ringed individual
+#'
+#' Assigns new unique identifiers to individuals that appear with two or more
+#' different ring numbers in a data set. The resulting table can then be used
+#' to replace all alternative ring numbers of an individual with the new
+#' identifier in all tables contained in the standard format.
+#'
+#' @param raw_data Data frame. Contains pairs of rings (one pair per
+#' re-ringing event). Column 'RING' = first ring, column 'RING2' = second ring.
+#' @param N Integer. The total number of rings carried by any one bird in the
+#' data.
+#'
+#' @return A data frame with columns 'RingNr' (= original ring number) and
+#' 'ReRingID' (= new unique identifier for the individual).
+#'
+#' @export
+#'
+#' @examples
+#' #Create fake dataset
+#' dat <- data.frame(RING = c('A1', 'B2', 'C7'), RING2 = c('B9', 'C7', 'E3'))
+#' #Summarise re-ringing information by individual
+#' make_ReRingTable(raw_data = dat, N = 3)
+
+make_ReRingTable = function(raw_data, N){
+
+  #We will be left joining one set of data to the other
+  #so we need to make sure the column names match
+  check_data  <- raw_data %>%
+    dplyr::rename("RING2" = .data$RING, "RING3" = .data$RING2)
+
+  #Create an output_data object that will be updated
+  output_data <- raw_data
+
+  #As long as any new ring values (e.g. RING2) are also used as an old ring value (e.g. RING)
+  #Then we need to keep going
+  #We use ncol() so that the code is robust to any number of re-ringings
+  while (any(check_data[, 2] %in% check_data[, 1])) {
+
+    #Add a new column to the output
+    #Instead of RING > RING2, we now have RING > RING2 > RING3
+    output_data <- output_data %>%
+      dplyr::left_join(check_data) %>%
```

Comment:
Add the by argument to prevent the 'joining by' message

```suggestion
      dplyr::left_join(check_data, by = "RING2") %>%
```